nmdc_automation/re_iding/scripts/re_id_tool.py

```diff
@@ -92,6 +91,219 @@ def cli(ctx, target):
+    orphan_metabolomics_activity_records = db_client["metabolomics_analysis_activity_set"].find(
+        {"id": {"$nin": non_orphan_metabolomics_ids}}
+    )
+    len_orphan_metabolomics_activity_records = len(list(orphan_metabolomics_activity_records.clone()))
+    logging.info(f"Found {len_orphan_metabolomics_activity_records} orphan Metabolomics Activity records")
+    if not no_update:
+        with session.start_transaction():
+            try:
+                _delete_metabolomics_orphans(updated_record_identifiers, orphan_metabolomics_activity_records, db_client)
+                session.commit_transaction()
+            except Exception as e:
+                logging.error(f"An error occurred - dumping updated record identifiers")
+                _write_updated_record_identifiers(updated_record_identifiers, "metabolomics")
+                session.abort_transaction()
+                logging.exception(f"An error occurred while deleting orphan records: {e} - aborting transaction")
+
+
+    _write_updated_record_identifiers(updated_record_identifiers, "metabolomics")
+    logging.info(f"Elapsed time: {time.time() - start_time}")
+
+
+def _delete_metabolomics_orphans(updated_record_identifiers, orphan_metabolomics_activity_records, db_client):
+    for orphan_metabolomics_activity_record in orphan_metabolomics_activity_records:
+        # get the data objects for has_input and has_output and delete them
+        has_input = orphan_metabolomics_activity_record.get("has_input", [])
+        for data_object_id in has_input:
+            updated_record_identifiers.append(("data_object_set", data_object_id, None))
+        result = db_client["data_object_set"].delete_many({"id": {"$in": has_input}})
+        logging.info(f"Deleted {result.deleted_count} data_object_set records")
+
+        has_output = orphan_metabolomics_activity_record.get("has_output", [])
+        for data_object_id in has_output:
+            updated_record_identifiers.append(("data_object_set", data_object_id, None))
+        result = db_client["data_object_set"].delete_many({"id": {"$in": has_output}})
+        logging.info(f"Deleted {result.deleted_count} data_object_set records")
+
+        logging.info(f"Deleting orphan Metabolomics Activity record: {orphan_metabolomics_activity_record['id']}")
+        updated_record_identifiers.append(
+            ("metabolomics_analysis_activity_set", orphan_metabolomics_activity_record["id"], None)
+            )
+        result = db_client["metabolomics_analysis_activity_set"].delete_one(
+            {"id": orphan_metabolomics_activity_record["id"]}
+            )
+        logging.info(f"Deleted {result.deleted_count} metabolomics_analysis_activity_set records")
+
+
+def _update_metabolomics_input_data_objects(input_data_objects, db_client, api_client, updated_record_identifiers):
+    """
+    Update the input data objects for a Metabolomics OmicsProcessing record.
```

Comment:
Minor but function comment should say "Metabolomics Workflow Activity records". The inputs to OmicsProcessing is a biosample or processed sample so the description doesn't make sense.

Reply:
https://github.com/microbiomedata/nmdc_automation/pull/116/commits/beebb0ae186d10adf9af0422bc228e7236426c18